core/chains/types.go

```diff
@@ -0,0 +1,18 @@
+package chains
+
+// ID types represent unique identifiers within a particular chain type. Using string is recommended.
+type ID any
+
+// Config types should have fields for chain configuration, and normally implement sql.Scanner and driver.Valuer, but
+// that is not enforced here since legacy types used mixed pointer and value receivers.
+type Config interface {
+	// sql.Scanner
+	// driver.Valuer
```

Comment:
I think this can be enforced soon - I'm just waiting for breaking and cross-repo changes until the end of this series.